lib/Sema/TypeCheckCircularity.cpp

```diff
@@ -601,3 +607,43 @@ bool CircularityChecker::diagnoseInfiniteRecursion(CanType parentType,
 
   return true;
 }
+
+// Show a warning if the enum is inconstructible. The outcome of this method
+// is irrelevant.
+void CircularityChecker::diagnoseInconstructible(EnumDecl *E) {
+
+  auto containsType = [](TupleType *tuple, Type E) -> bool {
+    for (auto type: tuple->getElementTypes()) {
+      if (type->isEqual(E))
+        return true;
+    }
+    return false;
+  };
+  auto isInconstructible = [containsType, E]() -> bool {
+    auto elts = E->getAllElements();
+    if (elts.empty())
+      return false;
+
+    for (auto elt: elts) {
+      if (!elt->hasInterfaceType() ||
```

Comment:
Could you assert that `RequireDelayedChecking` got tripped if the element has no interface type?

Reply:
It might not get tripped because I have a call to the method in the fast path before the loop in `expandEnum` where it is set.